pkg/clusteroperator/clusteroperator.go

```diff
@@ -0,0 +1,210 @@
+package clusteroperator
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "github.com/openshift/api/config/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	fiveScds time.Duration = 5 * time.Second
+	isTrue                 = "True"
+)
+
+// ClusterOperatorBuilder provides struct for ClusterOperator object.
+type ClusterOperatorBuilder struct {
+	Definition *v1.ClusterOperator
+	Object     *v1.ClusterOperator
+	apiClient  *clients.Settings
+	errorMsg   string
+}
+
+// Pull pulls existing ClusterOperator from cluster.
+func Pull(apiClient *clients.Settings, name string) (*ClusterOperatorBuilder, error) {
+	glog.V(100).Infof("Pulling existing ClusterOperator object name %s from cluster", name)
+
+	builder := ClusterOperatorBuilder{
+		apiClient: apiClient,
+		Definition: &v1.ClusterOperator{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the ClusterOperator is empty")
+
+		builder.errorMsg = "ClusterOperator 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("ClusterOperator object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Exists checks whether the given ClusterOperator exists.
+func (builder *ClusterOperatorBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if ClusterOperator %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get fetches the defined ClusterOperator from the cluster.
+func (builder *ClusterOperatorBuilder) Get() (*v1.ClusterOperator, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting ClusterOperator %s", builder.Definition.Name)
+
+	module := &v1.ClusterOperator{}
+
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
```

Comment:
You can use the client interface: builder.apiClient.ClusterOperators().Get()

Reply:
done